config/crd/bases/atlas.mongodb.com_atlasdatabaseusers.yaml

```diff
@@ -200,10 +214,13 @@ spec:
                 - CUSTOMER
                 type: string
             required:
-            - projectRef
             - roles
             - username
             type: object
+            x-kubernetes-validations:
+            - message: cannot define both atlasRef and projectRef at same time
+              rule: 'has(self.atlasRef) && !has(self.projectRef) ? true : !has(self.atlasRef)
+                && has(self.projectRef)'
```

Comment:
I am missing a unit test for this CEL.

As mentioned to @igor-karpukhin please be inspired by https://github.com/s-urbaniak/mongodb-atlas-kubernetes/commit/8b0af16a92c2809242f4c790191861111c7be79b#diff-86f3f6d950e0b76bc3ef5858a2280676cb73ead100b087d78270c714747a5249R134 to add one here.